### PR TITLE
buffer data vs payload app compat fix

### PR DESCRIPTION
**Pull request by pelikhan**
Adding macro to access "data" field.

### CODE REVIEW

File: libs/core/pxtbase.h [hunk 1]
```diff
@@ -7,6 +7,11 @@
 #pragma GCC diagnostic ignored "-Wformat"
 #pragma GCC diagnostic ignored "-Warray-bounds"
 
+// cross version compatible way of access data field
+#ifndef PXT_BUFFER_DATA
+#define PXT_BUFFER_DATA(buffer) buffer->data
+#endif
+
```
Comment (mmoskal) [inline libs/core/pxtbase.h]: This you will also need to add to `v0`

Comment (pelikhan) [reply libs/core/pxtbase.h]: i can add one in each package that needs it

Comment (mmoskal) [reply libs/core/pxtbase.h]: Right. When you're at it, maybe also add `#define PXT_MICROBIT_V1 true` so you have something to test on in the future

Comment (pelikhan) [reply libs/core/pxtbase.h]: Should I do something like #define PXT_MICROBIT_MAJOR_VERSION 1 ?

Comment (mmoskal) [reply libs/core/pxtbase.h]: well, then you need in the package:
```
#if PXT_MICROBIT_MAJOR_VERSION >= 1
...
```
which again, requires adding `#define PXT_MICROBIT_MAJOR_VERSION 0` to `v0`; but maybe that's the way to go?

Comment (pelikhan) [reply libs/core/pxtbase.h]: adding ```PXT_MICROBIT_TAGGED_INT```

**Pull request merged**